### ISSUE TITLE
Import Keras Tokenizer to Java Deeplearning4j (DL4J) for Text file classification

### ISSUE THREAD

**Issue by pavansolapure** [deeplearning4j/deeplearning4j#5889]
I have implemented the Text Classification of 20 News Group data using Keras (2.1.4 on TensorFlow). The accuracy is decent 0.87. I am also able to save the model and tokenizer and use them in another python program to predict the class of text file. I am using below to save model and tokenizer in python -

```
# creates a HDF5 file 'my_model.h5'
model.model.save('my_model.h5')
```
```
# Save Tokenizer i.e. Vocabulary
with open('tokenizer.pickle', 'wb') as handle:
    pickle.dump(tokenizer, handle, protocol=pickle.HIGHEST_PROTOCOL)
```

If you need to refere complete python code - https://github.com/pavansolapure/opencodez-samples/tree/master/keras-text-classification

Now, I am looking to import Keras trained model and tokenizer into **Java Web Application**. Deeplearning4j provides an option to load Keras model as

`MultiLayerNetwork network = KerasModelImport.importKerasSequentialModelAndWeights("PATH TO YOUR H5 FILE")`

But I could not find any option to load Tokenizer or its metadata.

Appreciate if DL4J plans to add support for this issue?

**Comment by maxpumperla** [comment]
@pavansolapure we have started a first attempt at importing preprocessing steps / data pipelines here: https://github.com/deeplearning4j/etl. This is still early stages however, and your case isn't covered. In any case, pickling your tokenizer will not be the solution (no way to unpickle in Java), but maybe we find a conventient way to add the tokenizer to an HDF5 group. We'll see.

**Comment by agibsonccc** [comment]
@pavansolapure What kind of tokenizer are you looking to map? This library can load pickle objects: https://github.com/irmen/Pyrolite that can handle things like vocab and the like. You might be able to manually map the tokenizer to something we have in dl4j already though.

**Comment by maxpumperla** [comment]
@pavansolapure just a quick update on this. We're working on the ability to properly serialize Keras preprocessors here: https://github.com/keras-team/keras-preprocessing/pull/29#pullrequestreview-141943118

At the same time I'm implementing Keras Tokenizers etc in Java and will support import as follow up, should the above PR get accepted. Otherwise serde will have to move to another project, which is also fine.

**Comment by maxpumperla** [comment]
@pavansolapure this is now supported in 1.0.0-beta2, see here:

https://github.com/deeplearning4j/deeplearning4j/blob/master/deeplearning4j/deeplearning4j-modelimport/src/test/java/org/deeplearning4j/nn/modelimport/keras/preprocessing/text/TokenizerImportTest.java

you probably have to install `keras-preprocessing` from master though. closing.

**Comment by pavansolapure** [comment]
Appreciate!
I will try this for sure.

**Comment by maxpumperla** [comment]
cool, happy to hear your feedback
